cpp/tests/preprocessing/spectral_embedding.cu

```diff
@@ -0,0 +1,243 @@
+    auto h_embedding =
+      raft::make_host_matrix<T, int, raft::col_major>(embedding_.extent(0), embedding_.extent(1));
+    raft::update_host(h_embedding.data_handle(),
+                      embedding_.data_handle(),
+                      embedding_.extent(0) * embedding_.extent(1),
+                      stream);
+    raft::resource::sync_stream(handle, stream);
+
+    bool all_zeros = true;
+    bool has_nan   = false;
+
+    for (int i = 0; i < h_embedding.extent(0) * h_embedding.extent(1); i++) {
+      if (h_embedding.data_handle()[i] != 0) { all_zeros = false; }
+      if (std::isnan(h_embedding.data_handle()[i])) {
+        has_nan = true;
+        break;
+      }
+    }
+
+    ASSERT_FALSE(all_zeros) << "Embedding contains all zeros";
+    ASSERT_FALSE(has_nan) << "Embedding contains NaN values";
+
+    // 3. Compute trustworthiness score to evaluate the quality of embedding
+    // Create views with int64_t indexing for trustworthiness_score function
+    auto input_view_i64 = raft::make_device_matrix_view<const float, int64_t, raft::row_major>(
+      input_.data_handle(), input_.extent(0), input_.extent(1));
+
+    // Need to convert embedding to row-major format for trustworthiness calculation
+    auto embedding_row_major = raft::make_device_matrix<float, int, raft::row_major>(
+      handle, embedding_.extent(0), embedding_.extent(1));
+
+    // Copy and transpose from col-major to row-major
+    raft::copy(embedding_row_major.data_handle(),
+               embedding_.data_handle(),
+               embedding_.extent(0) * embedding_.extent(1),
+               stream);
+
+    auto embedding_view_i64 = raft::make_device_matrix_view<const float, int64_t, raft::row_major>(
+      embedding_row_major.data_handle(),
+      embedding_row_major.extent(0),
+      embedding_row_major.extent(1));
+
+    // Calculate trustworthiness score
+    double trust_score = cuvs::stats::trustworthiness_score(
+      handle,
+      input_view_i64,
+      embedding_view_i64,
+      params_.n_neighbors,  // use same number of neighbors as the embedding
+      cuvs::distance::DistanceType::L2SqrtUnexpanded);
+
+    // Print the trustworthiness score (for reporting purposes)
+    std::cout << "Trustworthiness score: " << trust_score << std::endl;
+
+    // Check that trustworthiness score is reasonable (should be between 0 and 1, with higher being
+    // better)
+    ASSERT_GE(trust_score, 0.0) << "Trustworthiness score should be non-negative";
+    ASSERT_LE(trust_score, 1.0) << "Trustworthiness score should be <= 1.0";
+
+    // For good embeddings, we expect a reasonably high trustworthiness score
+    // This threshold can be adjusted based on empirical results
+    ASSERT_GT(trust_score, 0.7)
+      << "Trustworthiness score is too low, indicating poor embedding quality";
+
+    // Optional: Save the embedding for visualization
+    /*
+    std::ofstream out_file("embedding_output.csv");
```

Comment:
If you intend to keep this, I would suggest making this a function and accepting an argument to determine whether this should be done. We shouldn't keep commented out code in the codebase because it becomes very confusing to maintain as this code passes through different hands.